package/management/commands/pypi_update_download_package_stats.py

```diff
@@ -0,0 +1,95 @@
+import djclick as click
+import gzip
+import httpx
+import shutil
+
+from package.models import Package
+from pathlib import Path
+from rich import print
+from sqlite_utils import Database
+
+
+def decompress_gz_file(filename: str):
+    if not filename.endswith(".gz"):
+        print("File is not a .gz file")
+        return
+
+    with gzip.open(filename, "rb") as f_in:
+        with open(filename[:-3], "wb") as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+    print("File decompressed successfully")
+
+
+def download_file(*, url: str, filename: str):
+    if Path(filename).exists():
+        print("File already exists")
+        return
+
+    with httpx.get(url, stream=True) as r:
+        r.raise_for_status()
+        with open(filename, "wb") as f:
+            for chunk in r.iter_content(
+                chunk_size=8192
+            ):  # Choose the right chunk size for you
+                if chunk:  # filter out keep-alive new chunks
+                    f.write(chunk)
+
+    print("File downloaded successfully")
+
+
+def normalize_pypi_slug(url: str) -> str:
+    url = url.replace("http:", "https:")
+    url = url.strip("/")
+    url = url.split("/")[-1]
+    return url
+
+
+@click.command()
+@click.argument("url", type=str)
+@click.argument("filename", type=str)
```

Comment:
Do we need this argument?  Seems like we are getting the filename from the `url` argument. i.e:

`filename = normalize_pypi_slug(url=url)`